google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/AdminClientImpl.java

```diff
@@ -188,6 +194,29 @@ public CloudRegion region() {
                 .build());
   }
 
+  @Override
+  public OperationFuture<SeekSubscriptionResponse, OperationMetadata> seekSubscription(
+      SubscriptionPath path, SeekTarget target) {
+    SeekSubscriptionRequest.Builder request =
+        SeekSubscriptionRequest.newBuilder().setName(path.toString());
+    switch (target.getKind()) {
+      case BACKLOG_LOCATION:
+        if (target.backlogLocation() == BacklogLocation.END) {
+          request.setNamedTarget(NamedTarget.HEAD);
+        } else if (target.backlogLocation() == BacklogLocation.BEGINNING) {
```

Comment:
make this a nested switch instead to ensure full coverage (the compiler will fail if you don't handle all cases)

Reply:
Done. Thanks Dan.